toolchain/codegen/codegen.cpp

```diff
@@ -0,0 +1,72 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/codegen/codegen.h"
+
+#include <cstdio>
+
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/MC/TargetRegistry.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Target/TargetMachine.h"
+#include "llvm/Target/TargetOptions.h"
+#include "llvm/TargetParser/Host.h"
+
+namespace Carbon {
+
+auto PrintAssemblyFromModule(llvm::Module& module,
+                             llvm::StringRef target_triple,
+                             llvm::raw_pwrite_stream& error_stream,
+                             llvm::raw_pwrite_stream& output_stream) -> bool {
+  bool has_errors = false;
```

Comment:
Why are you creating a `has_errors` variable which is never assigned, and returning `has_errors`/`!has_errors`? Why not instead return true/false?